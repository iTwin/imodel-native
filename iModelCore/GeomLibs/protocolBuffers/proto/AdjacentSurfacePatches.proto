message AdjacentSurfacePatches
{
repeated ISurfacePatch Patch = 1;
}

