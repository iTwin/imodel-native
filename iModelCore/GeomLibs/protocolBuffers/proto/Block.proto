message Block
{
required IPlacement placement = 1;
required DPoint3d cornerA = 2;
required DPoint3d cornerB = 3;
required bool bSolidFlag = 4;
}

