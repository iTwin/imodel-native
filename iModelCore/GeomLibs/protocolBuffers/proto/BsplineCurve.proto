message BsplineCurve
{
required int32 Order = 1;
required bool Closed = 2;
repeated DPoint3d ControlPoint = 3;
repeated double Weight = 4;
repeated double Knot = 5;
}

