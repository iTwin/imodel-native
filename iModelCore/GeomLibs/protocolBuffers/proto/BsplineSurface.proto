message BsplineSurface
{
required int32 OrderU = 1;
required bool ClosedU = 2;
required int32 NumUControlPoint = 3;
required int32 OrderV = 4;
required bool ClosedV = 5;
required int32 NumVControlPoint = 6;
repeated DPoint3d ControlPoint = 7;
repeated double Weight = 8;
repeated double KnotU = 9;
repeated double KnotV = 10;
}

