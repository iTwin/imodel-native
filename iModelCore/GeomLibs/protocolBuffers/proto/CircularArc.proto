message CircularArc
{
required IPlacement placement = 1;
required double radius = 2;
required Angle startAngle = 3;
required Angle sweepAngle = 4;
}

