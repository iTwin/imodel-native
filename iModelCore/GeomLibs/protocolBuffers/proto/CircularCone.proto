message CircularCone
{
required IPlacement placement = 1;
required double height = 2;
required double radiusA = 3;
required double radiusB = 4;
required bool bSolidFlag = 5;
}

