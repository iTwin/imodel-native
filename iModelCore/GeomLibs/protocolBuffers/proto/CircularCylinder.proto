message CircularCylinder
{
required IPlacement placement = 1;
required double height = 2;
required double radius = 3;
required bool bSolidFlag = 4;
}

