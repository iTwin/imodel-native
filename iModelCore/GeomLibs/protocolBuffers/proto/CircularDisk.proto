message CircularDisk
{
required IPlacement placement = 1;
required double radius = 2;
}

