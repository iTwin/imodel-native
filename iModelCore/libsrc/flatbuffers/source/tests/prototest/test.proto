// Sample .proto file that we can translate to the corresponding .fbs.

package proto.test;

option some_option = is_ignored;
import "some_other_schema.proto";

enum ProtoEnum {
  FOO = 1;
  BAR = 5;
}

message OtherMessage {
  optional double a = 26;
  optional float b = 32 [default = 3.14149];
}

message ProtoMessage {
  optional int32 c = 12 [default = 16];
  optional int64 d = 1 [default = 0];
  optional uint32 p = 1;
  optional uint64 e = 2;
  optional sint32 f = 3 [default = -1];
  optional sint64 g = 4;
  optional fixed32 h = 5;
  optional fixed64 q = 6;
  optional sfixed32 i = 7;
  optional sfixed64 j = 8;
  optional bool k = 9;
  required string l = 10;
  optional bytes m = 11;
  optional OtherMessage n = 12;
  repeated string o = 14;
}
