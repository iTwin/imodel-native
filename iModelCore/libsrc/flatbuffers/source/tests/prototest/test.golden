// Generated from test.proto

namespace proto.test;

enum ProtoEnum : short {
  FOO = 1,
  BAR = 5,
}

table OtherMessage {
  a:double;
  b:float = 3.14149;
}

table ProtoMessage {
  c:int = 16;
  d:long;
  p:uint;
  e:ulong;
  f:int = -1;
  g:long;
  h:uint;
  q:ulong;
  i:int;
  j:long;
  k:bool;
  l:string (required);
  m:string;
  n:OtherMessage;
  o:[string];
}

